docs/docs/30-administration/22-backends/40-kubernetes.md

```diff
@@ -80,9 +80,32 @@ See the [kubernetes documentation](https://kubernetes.io/docs/concepts/security/
 
 ### nodeSelector
 
-Specify the label which is used to select the node where the job should be executed. Labels defined here will be appended to a list already containing "kubernetes.io/arch".
-By default the pod will use "kubernetes.io/arch" inferred from top-level "platform" setting which is deducted from the agents' environment variable CI_SYSTEM_PLATFORM. To overwrite this, you need to specify this label in the nodeSelector section.
-See the [kubernetes documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for more information on using nodeSelector.
+Specifies the label which is used to select the node on which the job will be executed.
+
+Labels defined here will be appended to a list which already contains `"kubernetes.io/arch"`.
+`"kubernetes.io/arch"` is inferred from the agents' environment variable `CI_SYSTEM_PLATFORM`.
+Without a manual overwrite, builds will be randomly assigned to the runners and inherit their respective architectures.
```

Comment:
This and below should be more concrete and placed in [Codeberg-CI/examples](https://codeberg.org/Codeberg-CI/examples) or in the separate `examples` directory here, I think.

Reply:
What do you mean by "more concrete"? I think having a short code example directly here is fine, https://codeberg.org/Codeberg-CI/examples is more for full pipelines.

Reply:
Now you have only two small pieces of your pipeline, but make statements like `builds will be randomly assigned` and `one must define`. This is not so obvious, not always true and you didn't provide full config when it's true. Each part is described well enough in its own page. If you want to show some use case when they are combined, you should show full pipeline. Of course, it should be as much as possible short, but full. Good examples are [1](https://woodpecker-ci.org/docs/next/usage/matrix-workflows#example-matrix-pipeline-using-multiple-platforms) and the last example from [comment](https://github.com/woodpecker-ci/woodpecker/pull/2531#issuecomment-1757460570).

Though, placing in Codeberg-CI/examples or in the separate examples directory might be overkill. Depends on pipeline length and specificity.

Reply:
> This is not so obvious, not always true and you didn't provide full config when it's true.

Hmm. It's the default which is inferred by "Without a manual overwrite,". When is it not true? Builds will be assigned to runners based on resource needs and what the scheduler thinks it's best - no? So each build could go to a different arch - it might not be explicitly random but from a user perspective it is.

> If you want to show some use case when they are combined, you should show full pipeline

Hm, I don't agree here. I think this is a style question.